sql/core/src/test/scala/org/apache/spark/sql/DatasetAggregatorSuite.scala

```diff
@@ -206,4 +216,12 @@ class DatasetAggregatorSuite extends QueryTest with SharedSQLContext {
       ds.groupByKey(_.b).agg(ComplexBufferAgg.toColumn),
       ("one", 1), ("two", 1))
   }
+
+  test("SPARK-12555 - data is not corrupted after input data is reordered") {
```

Comment:
I'm a little confused by this test name, what does `input data is reordered` means?